pkg/api/controller_test.go

```diff
@@ -782,3 +806,165 @@ func TestBasicAuthWithLDAP(t *testing.T) {
 		So(resp.StatusCode(), ShouldEqual, 200)
 	})
 }
+
+func TestBearerAuth(t *testing.T) {
+	Convey("Make a new controller", t, func() {
+		authTestServer := makeAuthTestServer()
+		defer authTestServer.Close()
+
+		config := api.NewConfig()
+		config.HTTP.Port = SecurePort3
+
+		u, err := url.Parse(authTestServer.URL)
+		So(err, ShouldBeNil)
+
+		config.HTTP.Auth = &api.AuthConfig{
+			Bearer: &api.BearerConfig{
+				Cert:    ServerCert,
+				Realm:   authTestServer.URL + "/auth/token",
+				Service: u.Host,
+			},
+		}
+		c := api.NewController(config)
+		dir, err := ioutil.TempDir("", "oci-repo-test")
+		So(err, ShouldBeNil)
+		defer os.RemoveAll(dir)
+		c.Config.Storage.RootDirectory = dir
+		go func() {
+			// this blocks
+			if err := c.Run(); err != nil {
+				return
+			}
+		}()
+
+		// wait till ready
+		for {
+			_, err := resty.R().Get(BaseURL3)
+			if err == nil {
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
+
+		defer func() {
+			ctx := context.Background()
+			_ = c.Server.Shutdown(ctx)
+		}()
+
+		blob := []byte("hello, blob!")
+		digest := godigest.FromBytes(blob).String()
+
+		resp, err := resty.R().Post(BaseURL3 + "/v2/" + AuthorizedNamespace + "/blobs/uploads/")
+		So(err, ShouldBeNil)
+		So(resp, ShouldNotBeNil)
+		So(resp.StatusCode(), ShouldEqual, 401)
+
+		authorizationHeader := parseBearerAuthHeader(resp.Header().Get("Www-Authenticate"))
+		resp, err = resty.R().
+			SetQueryParam("service", authorizationHeader.Service).
+			SetQueryParam("scope", "repository:"+AuthorizedNamespace+":push,pull").
```

Comment:
I believe if the zot auth layer is working properly, this should just be
```
SetQueryParam("scope", authorizationHeader.Scope)
```

(and below also). May need to also get another token fro the GET request (may have different scopes, push vs. pull)